src/irr/asset/CImageWriterPNG.cpp

```diff
@@ -156,12 +155,57 @@ bool CImageWriterPNG::writeAsset(io::IWriteFile* _file, const SAssetWriteParams&
 				return false;
 			}
 	}
+
+	core::smart_refctd_ptr<ICPUImage> newConvertedImage;
+	{
+		auto copyImageForConverting = core::smart_refctd_ptr_static_cast<ICPUImage>(image->clone());
+		auto copyImageParams = copyImageForConverting->getCreationParameters();
+		auto copyBuffer = copyImageForConverting->getBuffer();
+		auto copyRegion = copyImageForConverting->getRegions().begin();
+
+		auto newCpuBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(copyBuffer->getSize());
+		memcpy(newCpuBuffer->getPointer(), copyBuffer->getPointer(), newCpuBuffer->getSize());
+
+		auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1u);
+		ICPUImage::SBufferCopy& region = newRegions->front();
+		region.imageSubresource.mipLevel = copyRegion->imageSubresource.mipLevel;
+		region.imageSubresource.baseArrayLayer = copyRegion->imageSubresource.baseArrayLayer;
+		region.imageSubresource.layerCount = copyRegion->imageSubresource.layerCount;
+		region.bufferOffset = copyRegion->bufferOffset;
+		region.bufferRowLength = copyRegion->bufferRowLength;
+		region.bufferImageHeight = copyRegion->bufferImageHeight;
+		region.imageOffset = copyRegion->imageOffset;
+		region.imageExtent = copyRegion->imageExtent;
+
+		CMatchedSizeInOutImageFilterCommon::state_type state;
+		state.extent = imageParams.extent;
+
+		switch (imageParams.format)
+		{
+			case asset::EF_R8_UNORM:
+			{
+				copyImageParams.format = EF_R8_SRGB;
+				newConvertedImage = ICPUImage::create(std::move(copyImageParams));
+				newConvertedImage->setBufferAndRegions(std::move(newCpuBuffer), newRegions);
+				state.inImage = newConvertedImage.get();
+
+				CConvertFormatImageFilter<EF_R8_UNORM, EF_R8_SRGB> convertFiler;
+				convertFiler.execute(&state);
+			}
+			break;
+
+			default:
+			{
+				newConvertedImage = std::move(copyImageForConverting);
+			}
+			break;
+		}
+	}
```

Comment:
You want to to do this at the start, the exact same way as JPEG actually, except that you want one more option for `CConvertFilter<EF_UNKNOWN,EF_R8G8B8A8_SRGB>` for inputs with 4 channels, and only use `<EF_UNKNOWN,EF_R8G8B8_SRGB>` for 2 or 3 channel inputs

see https://github.com/Crisspl/IrrlichtBAW/pull/32/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h#r406725574